python/python/tests/test_blob.py

```diff
@@ -133,6 +133,46 @@ def test_blob_file_seek(tmp_path, dataset_with_blobs):
         assert f.read(1) == b"a"
 
 
+def test_null_blobs(tmp_path):
+    table = pa.table(
+        {
+            "id": range(100),
+            "blob": pa.array([None] * 100, pa.large_binary()),
+        },
+        schema=pa.schema(
+            [
+                pa.field("id", pa.uint64()),
+                pa.field(
+                    "blob", pa.large_binary(), metadata={"lance-encoding:blob": "true"}
+                ),
+            ]
+        ),
+    )
+    ds = lance.write_dataset(table, tmp_path / "test_ds")
+
+    blobs = ds.take_blobs("blob", ids=range(100))
+    for blob in blobs:
+        assert blob.size() == 0
+
+    ds.insert(pa.table({"id": pa.array(range(100, 200), pa.uint64())}))
+
+    blobs = ds.take_blobs("blob", indices=range(100, 200))
```

Comment:
Could you also add a test to cover `add_column` case?

https://github.com/lancedb/lance/issues/3840